comp/collector/collector/collectorimpl/collector.go

```diff
@@ -86,20 +93,30 @@ func Module() fxutil.Module {
 
 func newProvides(deps dependencies) provides {
 	c := newCollector(deps)
+
+	var agentCheckMetadata metadata.Provider
+	if _, isSet := deps.MetricSerializer.Get(); isSet {
+		agentCheckMetadata = metadata.NewProvider(c.collectMetadata)
+	}
```

Comment:
It seems that this component may end up providing nil Provider. Will all dependents be able to handle it?

Reply:
yes, we now filter groups on the user side rather than the provider side.